### PR TITLE
Fix minor issues found while reviewing task states PR's

### PR DESCRIPTION
**Pull request by kinow**
This is a small change with no associated Issue. Found one strange thing some code that was not in the PR I was reviewing (#3423) so ran a complete code analysis with the IDE, and selected the issues that seemed to be legit (it actually shows heaps more warnings, especially related to typing - for some other day).

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

**Comment by kinow** [comment]
There are two warnings that I think could be bugs, but I will need to debug with some test suite. If any of them turns out to be a bug, I'll add tests & changelog. Draft PR for now.

File: cylc/flow/parsec/validate.py [hunk 1]
```diff
@@ -909,7 +909,6 @@ def coerce_xtrigger(cls, value, keys):
         value = cls.strip_and_unquote(keys, value)
         if not value:
             raise IllegalValueError("xtrigger", keys, value)
-        fname = None
```
Comment (kinow) [inline cylc/flow/parsec/validate.py:912]: (click the blue down-arrow in GH UI to view `fname` declared further down)

File: cylc/flow/network/scan.py [hunk 2]
```diff
@@ -190,7 +190,7 @@ async def filter_name(flow, pattern):
     Args:
         flow (dict):
             Flow information dictionary, provided by scan through the pipe.
-        pattern (str):
+        pattern (re.Pattern):
```
Comment (kinow) [inline cylc/flow/network/scan.py:193]: See further down where we call `.match`, method from `re.Pattern` I think.

File: cylc/flow/remote.py [hunk 3]
```diff
@@ -223,15 +222,14 @@ def construct_rsync_over_ssh_cmd(
 
 
 def construct_ssh_cmd(
-        raw_cmd, user=None, host=None, forward_x11=False, stdin=False,
```
Comment (kinow) [inline cylc/flow/remote.py:226]: The `user` appears to be calculated in runtime, with no need to pass a `user` parameter. This `user` parameter is never used within `construct_ssh_cmd`, which can be misleading to a user/caller of the API.

File: cylc/flow/scheduler.py [hunk 4]
```diff
@@ -626,7 +625,7 @@ async def start_scheduler(self):
         finally:
             self.profiler.stop()
 
-    async def run(self, daemonize=False):
```
Comment (kinow) [inline cylc/flow/scheduler.py:629]: Probably removed some time ago. We still have the `daemonize` function elsewhere. This `daemonize` flag is never used here. Also doesn't seem to be used by the callers of this function.

File: cylc/flow/suite_files.py [hunk 5]
```diff
@@ -389,7 +389,7 @@ def get_suite_srv_dir(reg, suite_owner=None):
     return os.path.join(run_d, SuiteFiles.Service.DIRNAME)
 
 
-def load_contact_file(reg, owner=None, host=None):
```
Comment (kinow) [inline cylc/flow/suite_files.py:392]: Not used.

File: cylc/flow/task_state.py [hunk 6]
```diff
@@ -485,7 +485,6 @@ def _add_prerequisites(self, point, tdef):
 
         if tdef.sequential:
             # Add a previous-instance succeeded prerequisite.
-            p_prev = None
```
Comment (kinow) [inline cylc/flow/task_state.py:488]: See below for more, `p_prev` is re-declared only in the `if adjusted` branch.

File: cylc/flow/remote.py (hunk 3 above)
Comment (hjoliver) [reply cylc/flow/remote.py:226]: I think this was associated with the "task owner" concept we used to have (run a task job as user@host) but recently removed (now use ssh config instead if you need to run as a different user on host).

**Comment by kinow** [comment]
Last issue seems to be a bug but I'm still testing it. Will leave that one for later in a separate PR.

File: cylc/flow/xtriggers/suite_state.py [hunk 7]
```diff
@@ -58,9 +58,6 @@ def suite_state(suite, task, point, offset=None, status='succeeded',
                different location to what is specified in the global
                configuration (usually ``~/cylc-run``).
 
-        debug (bool):
-            Flag to enable debug information.
-
```
Comment (hjoliver) [inline cylc/flow/xtriggers/suite_state.py:63]: Can you remove `(except ''debug")` from the bottom of the docstring too? (line 71)

**Review by hjoliver — CHANGES_REQUESTED** [review]
(One tiny additional change suggested)

File: cylc/flow/xtriggers/suite_state.py (hunk 7 above)
Comment (kinow) [reply cylc/flow/xtriggers/suite_state.py:63]: Done!

**Comment by hjoliver** [comment]
One review will do for this @kinow - thanks :bouquet:

**Pull request merged**